src/services/mediaDevicePreferences.ts

```diff
@@ -63,10 +71,10 @@ ${videoInputList.map(getDeviceString).join('\n')}
  * @param devices list of available devices
  * @param inputList list of registered audio/video devices in order of preference
  * @param [fallbackId] id of currently selected input
- * @return first available (plugged) device id or fallback
+ * @return {string|undefined} first available (plugged) device id or fallback
  */
 function getFirstAvailableMediaDevice(devices: MediaDeviceInfo[], inputList: MediaDeviceInfo[], fallbackId?: string): string | undefined {
-	const availableDevices = devices.map(device => device.deviceId).filter(id => id !== 'default')
+	const availableDevices = devices.map(device => device.deviceId).filter(id => id !== DEFAULT_ID)
 	return inputList.find(device => availableDevices.includes(device.deviceId))?.deviceId ?? fallbackId
 }
```

Comment:
fallbackId is just an intermediary and can be skipped, here you have a fall back and also in the function all itself. 
- We should have default device in the list 
- Do the loop:
    - If a plugged device found, done
    - If not ( including default), return the value that indicates that no device is selected ( I am not sure if it's `null` or `undefined`) 
    - remove the fallback on the function call 
    `this.attributes.audioInputId = getFirstAvailableMediaDevice(devices, this._preferenceAudioInputList, this._fallbackAudioInputId) ?? 'default'`

and by having all devices in the list,  it means that we have the fallback there too.